homeassistant/components/opentherm_gw/sensor.py

```diff
@@ -1,20 +1,621 @@
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_SETPOINT,
+            friendly_name_format="Hot Water Setpoint {}",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_MAX_CH_SETPOINT,
+            friendly_name_format="Maximum Central Heating Setpoint {}",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_OEM_DIAG,
+            friendly_name_format="OEM Diagnostic Code {}",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_TOTAL_BURNER_STARTS,
+            friendly_name_format="Total Burner Starts {}",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_CH_PUMP_STARTS,
+            friendly_name_format="Central Heating Pump Starts {}",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_PUMP_STARTS,
+            friendly_name_format="Hot Water Pump Starts {}",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_DHW_BURNER_STARTS,
+            friendly_name_format="Hot Water Burner Starts {}",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement="starts",
+        ),
+    ),
+    (
+        [gw_vars.BOILER, gw_vars.THERMOSTAT],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.DATA_TOTAL_BURNER_HOURS,
+            friendly_name_format="Total Burner Hours {}",
+            device_class=SensorDeviceClass.DURATION,
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        ),
+    ),
```

Comment:
Does this only increase? Should this use total_increasing?

Reply:
It normally only increases, but it wraps after `65535`. Some boilers allow resetting the value to `0`, but support for this is not mandatory (and we don't provide a mechanism for it in HA apart from sending a raw protocol command through `SERVICE_SEND_TRANSP_CMD`). I chose `SensorStateClass.TOTAL` as per the [dev docs](https://developers.home-assistant.io/docs/core/entity/sensor#entities-representing-a-total-amount) for a value that only increases.